810 me0 geometry pr v1

pietverwilligen opened the pull request:
This is the first of two pull requests for an improved (realistic) readout of the ME0 detector. This pull request enables the use of an me0.xml file introduced earlier by Sunanda in which the ME0 detector is split in 10 eta partitions (#14537). In a forthcoming pr we will include realistic digitization (w.r.t. the pseudodigitization included in the release). We would like to stress that for the muon upgrade simulations this PR his highly desired to enter into the 810 release, please test. Thanks.

NB I tried both:
scram b vclean && scram b -k -j 48
scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'
with a good outcome.

cmsbuild commented:
A new Pull Request was created by  @pietverwilligen (Piet Verwilligen) for CMSSW_8_1_X.

It involves the following packages:

Configuration/Geometry
DataFormats/MuonDetId
Geometry/CMSCommonData
Geometry/GEMGeometry
Geometry/GEMGeometryBuilder
Geometry/MuonCommonData
Geometry/MuonNumbering
Geometry/MuonSimData


@civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @battibass, @abbiendi, @jhgoh, @Martin-Grunewald, @calderona, @HuguesBrun, @ptcox, @trocino this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

pietverwilligen commented:
Can you please have a look kpedro88?
Thanks.

cmsbuild commented:
Pull request #16114 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

pietverwilligen commented:
forgot to supress a debug flag. fixed now. please test.

civanch commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 commented:
We are no longer making 2023 geometries by hand, so `Extended2023D1ME0Dev` is not acceptable. Please go to [Configuration/Geometry/scripts/dict2023Geometry.py](https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/Geometry/scripts/dict2023Geometry.py) and add "M2" to `muonDict` with the appropriate changes. Then you can add something like `("O1","T1","C1","M2","F1") : "D5",` in `detectorVersionDict` and run the script (1 is the default version for each subdetector, so only non-1 values need to be specified):
```
python generate2023Geometry.py -M 2
```
You should also update the README accordingly.

ianna commented on Configuration/Geometry/python/GeometryExtended2023D1ME0Dev_cff.py:
```diff
@@ -0,0 +1,9 @@
+import FWCore.ParameterSet.Config as cms
```
@pietverwilligen - this file should be removed. It will become an auto-generated GeometryExtended2023D5 after you add D5 in the dictionary and run the script to generate it. This is valid for all new cff and cfi files added in this PR.

cmsbuild commented:
-1

Tested at: 3b455c805f7309655468c4727bf8835b6160820d


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/a454bda5f242bb2fb62c73da098c4be3584e24c7
https://github.com/cms-sw/cmssw/commit/51e36d14a5be387d1e3937e14f4cc6da5a766bd6
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test testRecoMETMETProducers had ERRORS



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/a454bda5f242bb2fb62c73da098c4be3584e24c7
https://github.com/cms-sw/cmssw/commit/51e36d14a5be387d1e3937e14f4cc6da5a766bd6
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/summary.html

cmsbuild commented:
Pull request #16114 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @franzoni, @davidlange6 can you please check and sign again.

pietverwilligen commented:
Thanks for pointing me to the relevant information. I removed 2023D1ME0Dev and created 2023D5 geometry and updated the READme accordingly.

Wrt to the failed unitTests, I believe this is something not related to this PR, but a problem opening a rootfile. Can some experts have a look and express an opinion? Thanks!

Looking into:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15564/unitTests.log

and searching for 
testRecoMETMETProducers

I find:
06-Oct-2016 17:02:36 CEST  Initiating request to open file root://eoscms.cern.ch//eos/cms/store/relval/CMSSW_7_3_0_pre1/RelValZEE_13/GEN-SIM-RECO/PRE_LS172_V15-v1/00000/C2212C6F-EB59-E411-AB78-0025905B85EE.root
06-Oct-2016 17:02:48 CEST  Successfully opened file root://eoscms.cern.ch//eos/cms/store/relval/CMSSW_7_3_0_pre1/RelValZEE_13/GEN-SIM-RECO/PRE_LS172_V15-v1/00000/C2212C6F-EB59-E411-AB78-0025905B85EE.root
Begin processing the 1st record. Run 1, Event 101, LumiSection 2 at 06-Oct-2016 17:03:01.301 CEST
06-Oct-2016 17:03:01 CEST  Closed file root://eoscms.cern.ch//eos/cms/store/relval/CMSSW_7_3_0_pre1/RelValZEE_13/GEN-SIM-RECO/PRE_LS172_V15-v1/00000/C2212C6F-EB59-E411-AB78-0025905B85EE.root

TrigReport ---------- Event  Summary ------------
TrigReport Events total = 10 passed = 10 failed = 0

TrigReport ---------- Path   Summary ------------
TrigReport  Trig Bit#   Executed     Passed     Failed      Error Name
TrigReport     1    0         10         10          0          0 p

TrigReport -------End-Path   Summary ------------
TrigReport  Trig Bit#   Executed     Passed     Failed      Error Name
TrigReport     0    0         10         10          0          0 e1

[...]
T---Report end!
=============================================

MessageLogger Summary

 type     category        sev    module        subroutine        count    total
 ---- -------------------- -- ---------------- ----------------  -----    -----
    1 fileAction           -s file_close                             1        1
    2 fileAction           -s file_open                              2        2

 type    category    Examples: run/evt        run/evt          run/evt
 ---- -------------------- ---------------- ---------------- ----------------
    1 fileAction           PostEndRun                        
    2 fileAction           pre-events       pre-events       

Severity    # Occurrences   Total Occurrences
--------    -------------   -----------------
System                  3                   3
06-Oct-2016 17:03:15 CEST  Initiating request to open file file:ttbarForMetTests.root
----- Begin Fatal Exception 06-Oct-2016 17:03:15 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootFileSequenceBase::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling File::sysopen()
Exception Message:
Failed to open the file 'ttbarForMetTests.root'
   Additional Info:
      [a] Input file file:ttbarForMetTests.root could not be opened.
      [b] open() failed with system error 'No such file or directory' (error code 2)
----- End Fatal Exception -------------------------------------------------
Failure using recoMET_caloMet_cfg.py: status 84
status = 21504

---> test testRecoMETMETProducers had ERRORS
 
^^^^ End Test testRecoMETMETProducers ^^^^

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 commented:
@pietverwilligen The file appears to be missing because of some error in Pythia:
```
PYTHIA Error in Pythia::check: charge not conserved  
 PYTHIA Abort from Pythia::forceHadronLevel: check of event revealed problems  
 PYTHIA Warning in Pythia::check: not quite matched particle energy/momentum/mass  
 PYTHIA Error in Pythia::check: unmatched particle energy/momentum/mass  
```

I'm not sure how this PR would have affected Pythia, so maybe it is spurious or caused by a different PR.

cmsbuild commented:
+1
Tested at: 00bb024f048007fa1fb009b51b0a2ae8b86c6f61
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15599/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16114/15599/summary.html

ianna commented:
@pietverwilligen - please, rebase

cmsbuild commented:
Pull request #16114 was updated. @ghellwig, @ianna, @cerminar, @fabozzi, @rekovic, @vanbesien, @civanch, @monttj, @franzoni, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @dmitrijus, @alja, @srimanob, @hengne can you please check and sign again.

pietverwilligen commented:
PR rebased to CMSSW_8_1_X_2016-10-11-1100

kpedro88 commented:
@pietverwilligen did you actually rebase or did you merge? This PR now has 175 commits.

pietverwilligen commented:
I issued "merge" as final command, i should have issued "rebase"? How can I undo?

kpedro88 commented:
Rebase has a specific meaning in git, and is very different from merge.

Here are instructions to rebase from one IB to another in CMSSW:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideHATS2016Git#CMSSW_rebasing_recipe

To undo what you previously did, go to your old IB working area and type `git reflog`. This will show you a list of previous states of your area, like this:
```
b785ac8 HEAD@{0}: commit: ...
882f8a4 HEAD@{1}: commit: ...
```
The first column is the hash of that state. Find the one before you merged, and type `git reset --hard [hash]`.

pietverwilligen commented:
Thx. I ll fix it asap.

On 11 Oct 2016 18:00, "Kevin Pedro" <notifications@github.com> wrote:

> Rebase has a specific meaning in git, and is very different from merge.
>
> Here are instructions to rebase from one IB to another in CMSSW:
> https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideHATS2016Git#CMSSW_
> rebasing_recipe
>
> To undo what you previously did, go to your old IB working area and type git
> reflog. This will show you a list of previous states of your area, like
> this:
>
> b785ac8 HEAD@{0}: commit: ...
> 882f8a4 HEAD@{1}: commit: ...
>
> The first column is the hash of that state. Find the one before you
> merged, and type git reset --hard [hash].
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16114#issuecomment-252961530>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFJgOZsyhobjSldjUgGi4j_OkTKFmtsxks5qy7J1gaJpZM4KPzio>
> .
>

pietverwilligen commented:
Hi @kpedro88, I tried to get my PR back before i issued the merge but i could not push it into the repository:

git reset --hard 146ef82
HEAD is now at 146ef82 Merge pull request #16081 from lgray/phase2_timing_geometry
git push my-cmssw 810_Geometry_PR_v1
Enter passphrase for key '/lustre/home/piet/.ssh/id_rsa': 
error: src refspec 810_Geometry_PR_v1 does not match any.
error: failed to push some refs to 'git@github.com:pietverwilligen/cmssw.git'

kpedro88 commented:
@pietverwilligen you have to do `git push -f` because you changed the history of the branch.

slava77 commented:
-1
needs a rebase or other steps to cleanup the commit history

cmsbuild commented:
Pull request #16114 was updated. @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @franzoni, @davidlange6 can you please check and sign again.

pietverwilligen commented:
hehe ... i did not want to close it but thats were we are now :-(. I am not even able to reopen it.

pietverwilligen commented:
i opened a new PR for these changes: #16185
